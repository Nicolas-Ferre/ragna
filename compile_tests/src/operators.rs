fn main() {}

#[ragna::gpu]
mod gpu {
    const CONSTANT: u32 = 10;

    static INVALID_NEG: u32 = -CONSTANT;
    static INVALID_NOT: u32 = !CONSTANT;
    static INVALID_DEREF: u32 = *&CONSTANT;
}
