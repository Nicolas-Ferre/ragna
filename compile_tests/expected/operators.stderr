error: unsupported unary operator
 --> compile_tests/src/operators.rs:9:33
  |
9 |     static INVALID_DEREF: u32 = *&CONSTANT;
  |                                 ^

error: unsupported expression
 --> compile_tests/src/operators.rs:9:34
  |
9 |     static INVALID_DEREF: u32 = *&CONSTANT;
  |                                  ^

error[E0277]: the trait bound `u32: ragna::operators::GpuNeg` is not satisfied
   --> compile_tests/src/operators.rs:7:31
    |
7   |     static INVALID_NEG: u32 = -CONSTANT;
    |                               -^^^^^^^^
    |                               |
    |                               the trait `ragna::operators::GpuNeg` is not implemented for `u32`
    |                               required by a bound introduced by this call
    |
    = help: the following other types implement trait `ragna::operators::GpuNeg`:
              f32
              i32

error[E0277]: the trait bound `u32: ragna::operators::GpuNot` is not satisfied
   --> compile_tests/src/operators.rs:8:31
    |
8   |     static INVALID_NOT: u32 = !CONSTANT;
    |                               -^^^^^^^^
    |                               |
    |                               the trait `ragna::operators::GpuNot` is not implemented for `u32`
    |                               required by a bound introduced by this call
    |
    = help: the trait `ragna::operators::GpuNot` is implemented for `bool`
